airflow/providers/snowflake/hooks/snowflake.py

```diff
@@ -216,7 +216,16 @@ def _get_conn_params(self) -> dict[str, str | None]:
                 "Please remove one."
             )
         elif private_key_file:
-            private_key_pem = Path(private_key_file).read_bytes()
+            private_key_file_path = Path(private_key_file)
+            if not private_key_file_path.is_file() or private_key_file_path.stat().st_size == 0:
+                raise AirflowException(
+                    "The private_key_file path points to an empty or invalid file."
+                )
```

Comment:
What happens if we don’t check for these?

Reply:
```Passing path on an excessively large file or path to a system device crashes Webserver. This fix validates the path against such situations.``` from the description of the PR.

Reply:
If we don't check if path is a file a system device can be passed. e.g. `/dev/random` That would also crash Webserver.

Reply:
Still does not explain the extra stat check though? `is_file` only returns true for regular files.

Also this can benefit from a better exception class than AirflowException. A ValueError, OSError, or PermissionError would be more appropriate IMO.

Reply:
That is the intention - the private key path should point to a regular file. Does it make sense to read the key from other type of files?

I'm ok with changing to `ValueError` - pushed the change.

Reply:
> the private key path should point to a regular file

And the `is_file()` call checks that. What does the other check cover?

Reply:
I checked also that for system `st_size` reports size as 0 therefore it's an alternative way of checking if this is a valid file. On top of that it doesn't make sense to pass an empty file in this case.

Reply:
I checked also that for system `st_size` reports size as 0 therefore it's an alternative way of checking if this is a valid file. On top of that it doesn't make sense to pass an empty file in this case.